### PR TITLE
QUAL rename facnumber by ref

### PR DESCRIPTION
**Pull request by atm-maxime**
No trouble at all with this. I checked every screen modified.
But now I think a little communication to the dev community is required... :smile:

### CODE REVIEW

File: htdocs/install/mysql/migration/9.0.0-10.0.0.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-- -- VPGSQL8.2 SELECT setval('llx_table_rowid_seq', MAX(rowid)) FROM llx_table;
+-- To set a field as NULL:                     -- VMYSQL4.3 ALTER TABLE llx_table MODIFY COLUMN name varchar(60) NULL;
+-- To set a field as NULL:                     -- VPGSQL8.2 ALTER TABLE llx_table ALTER COLUMN name DROP NOT NULL;
+-- To set a field as NOT NULL:                 -- VMYSQL4.3 ALTER TABLE llx_table MODIFY COLUMN name varchar(60) NOT NULL;
+-- To set a field as NOT NULL:                 -- VPGSQL8.2 ALTER TABLE llx_table ALTER COLUMN name SET NOT NULL;
+-- To set a field as default NULL:             -- VPGSQL8.2 ALTER TABLE llx_table ALTER COLUMN name SET DEFAULT NULL;
+-- Note: fields with type BLOB/TEXT can't have default value.
+
+
+ALTER TABLE llx_facture CHANGE facnumber ref VARCHAR(30) NOT NULL;
```
Comment (hregis) [inline htdocs/install/mysql/migration/9.0.0-10.0.0.sql]: @atm-maxime 
you forgot to delete and recreate the unique key "idx_facture_uk_facnumber" in "idx_facture_uk_ref"

Comment (atm-maxime) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql]: Thanks, done !

File: htdocs/install/mysql/migration/9.0.0-10.0.0.sql [hunk 2]
```diff
@@ -27,5 +27,5 @@
 -- To set a field as default NULL:             -- VPGSQL8.2 ALTER TABLE llx_table ALTER COLUMN name SET DEFAULT NULL;
 -- Note: fields with type BLOB/TEXT can't have default value.
 
-
-ALTER TABLE llx_facture CHANGE facnumber ref VARCHAR(30) NOT NULL;
\ No newline at end of file
+ALTER TABLE llx_facture CHANGE facnumber ref VARCHAR(30) NOT NULL;
+ALTER TABLE llx_facture DROP INDEX idx_facture_uk_facnumber, ADD UNIQUE idx_facture_uk_ref (ref, entity);
```
Comment (hregis) [inline htdocs/install/mysql/migration/9.0.0-10.0.0.sql]: @atm-maxime 
drop index before changing field name and recreate after

ALTER TABLE llx_facture DROP INDEX idx_facture_uk_facnumber;
ALTER TABLE llx_facture CHANGE facnumber ref VARCHAR(30) NOT NULL;
ALTER TABLE llx_facture ADD UNIQUE idx_facture_uk_ref (ref, entity);

Comment (eldy) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql]: Also, we must use 1 line for each action (no DROP..., ADD ... allowed by migration process)

**Comment by atm-maxime** [comment]
Done :)

**Comment by frederic34** [comment]
@atm-maxime do you need this https://github.com/Dolibarr/dolibarr/pull/10185 to fix travis?

**Pull request merged**